File: ZhigaoWeather/app/src/main/java/com/weather/zhigao/model/weather/BasicBean.java
package com.weather.zhigao.model.weather;

import android.os.Parcel;
import android.os.Parcelable;

public  class BasicBean implements Parcelable {
            /**
             * cid : CN101010300
             * location : 朝阳
             * parent_city : 北京
             * admin_area : 北京
             * cnty : 中国
             * lat : 39.92148972
             * lon : 116.48641205
             * tz : +8.00
             */

            public String cid;
            public String location;
            public String parent_city;
            public String admin_area;
            public String cnty;
            public String lat;
            public String lon;
            public String tz;

            public String getCid() {
                return cid;
            }

            public void setCid(String cid) {
                this.cid = cid;
            }

            public String getLocation() {
                return location;
            }

            public void setLocation(String location) {
                this.location = location;
            }

            public String getParent_city() {
                return parent_city;
            }

            public void setParent_city(String parent_city) {
                this.parent_city = parent_city;
            }

            public String getAdmin_area() {
                return admin_area;
            }

            public void setAdmin_area(String admin_area) {
                this.admin_area = admin_area;
            }

            public String getCnty() {
                return cnty;
            }

            public void setCnty(String cnty) {
                this.cnty = cnty;
            }

            public String getLat() {
                return lat;
            }

            public void setLat(String lat) {
                this.lat = lat;
            }

            public String getLon() {
                return lon;
            }

            public void setLon(String lon) {
                this.lon = lon;
            }

            public String getTz() {
                return tz;
            }

            public void setTz(String tz) {
                this.tz = tz;
            }

            @Override
            public int describeContents() {
                return 0;
            }

            @Override
            public void writeToParcel(Parcel dest, int flags) {
                dest.writeString(this.cid);
                dest.writeString(this.location);
                dest.writeString(this.parent_city);
                dest.writeString(this.admin_area);
                dest.writeString(this.cnty);
                dest.writeString(this.lat);
                dest.writeString(this.lon);
                dest.writeString(this.tz);
            }

            public BasicBean() {
            }

            protected BasicBean(Parcel in) {
                this.cid = in.readString();
                this.location = in.readString();
                this.parent_city = in.readString();
                this.admin_area = in.readString();
                this.cnty = in.readString();
                this.lat = in.readString();
                this.lon = in.readString();
                this.tz = in.readString();
            }

            public static final Creator<BasicBean> CREATOR = new Creator<BasicBean>() {
                @Override
                public BasicBean createFromParcel(Parcel source) {
                    return new BasicBean(source);
                }

                @Override
                public BasicBean[] newArray(int size) {
                    return new BasicBean[size];
                }
            };
        }

File: ZhigaoWeather/app/src/main/java/com/weather/zhigao/model/weather/BasicBean.kt
package com.weather.zhigao.model.weather

import android.os.Parcel
import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
class BasicBean(var cid: String,var location: String,var parent_city: String,var admin_area: String, var cnty: String,var lat: String, var lon: String,var tz: String) : Parcelable {
    /**
     * cid : CN101010300
     * location : 朝阳
     * parent_city : 北京
     * admin_area : 北京
     * cnty : 中国
     * lat : 39.92148972
     * lon : 116.48641205
     * tz : +8.00
     */

//    var cid: String
//    var location: String
//    var parent_city: String
//    var admin_area: String
//    var cnty: String
//    var lat: String
//    var lon: String
//    var tz: String
//
//    override fun describeContents(): Int {
//        return 0
//    }
//
//    override fun writeToParcel(dest: Parcel, flags: Int) {
//        dest.writeString(this.cid)
//        dest.writeString(this.location)
//        dest.writeString(this.parent_city)
//        dest.writeString(this.admin_area)
//        dest.writeString(this.cnty)
//        dest.writeString(this.lat)
//        dest.writeString(this.lon)
//        dest.writeString(this.tz)
//    }
//
//    constructor() {}
//
//    protected constructor(`in`: Parcel) {
//        this.cid = `in`.readString()
//        this.location = `in`.readString()
//        this.parent_city = `in`.readString()
//        this.admin_area = `in`.readString()
//        this.cnty = `in`.readString()
//        this.lat = `in`.readString()
//        this.lon = `in`.readString()
//        this.tz = `in`.readString()
//    }
//
//    companion object {
//
//        val CREATOR: Parcelable.Creator<BasicBean> = object : Parcelable.Creator<BasicBean> {
//            override fun createFromParcel(source: Parcel): BasicBean {
//                return BasicBean(source)
//            }
//
//            override fun newArray(size: Int): Array<BasicBean> {
//                return arrayOfNulls(size)
//            }
//        }
//    }
}